src/FocusTrap.ts

```diff
@@ -76,14 +78,30 @@ export const FocusTrap = defineComponent({
 
   setup(props, { slots, emit }) {
     let trap: FocusTrapI | null
-    const el = ref<HTMLElement | null>(null)
+    const wrapperElement = ref<
+      HTMLElement | ComponentPublicInstance | undefined
+    >()
+    const el = computed<HTMLElement>(() => {
+      if (!wrapperElement.value) return undefined
+
+      if (
+        !(wrapperElement.value instanceof HTMLElement) &&
+        wrapperElement.value?.$el
+      ) {
+        return (wrapperElement.value as ComponentPublicInstance).$el
+      } else if (wrapperElement.value instanceof HTMLElement) {
+        return wrapperElement.value
+      }
+      return undefined
+    })
```

Comment:
This should be something like this (needs formatting:

```suggestion
    const el = computed<HTMLElement | undefined>(() =>  {
      const element = wrapperElement.value
      return element && (element instanceof HTMLElement ? element : element.$el)
    })
```

Reply:
I missed out on the `| undefined` type in the latest commit, and yes it should be there, at least ideally.

This introduces a problem, however; if we want to ensure that the error message will only show up in `process.env.NODE_ENV === 'development'` and not production, then that begs for a condition like `__DEV__ && el.value !== undefined`. TypeScript isn't happy about that, as it should, as the consequent use of `el.value` in `ensureTrap` is invalid (in particular, `undefined` is not an allowed type). In particular, type narrowing will not be applicable, as `__DEV__ && el.value` evaluates to `false` when `__DEV__ === false`, and so `el.value` can indeed be `undefined`.